### ISSUE TITLE
XYZ + Matrix Profile Support

### ISSUE THREAD

**Issue by theaspiringitpro** [ledoge/novideo_srgb#13]
@ledoge, I noticed the readme was updated. Does this tool now support XYZ + Matrix profiling?

I did a little testing and chose a XYZ + Matrix profile but am getting quite odd results. Everything on the desktop has a red hue on it now. Doesn't seem to be working properly.

**Comment by ledoge** [comment]
Profiles with XYZ LUTs *should* work properly now, but I only tested it with one so there could still be some bugs. It'll use that in place of the curve(s) + matrix data.

When you try a previous version of novideo_srgb (which will use the curve + matrix data), do you get the same results? Also, please upload the profile so I can take a closer look.

**Comment by theaspiringitpro** [comment]
Here's a screenshot of what I'm seeing. Previous versions using a 'curves + matrix' profile worked just fine.

![screenshot](https://user-images.githubusercontent.com/16659780/150608521-b273f41b-8289-4781-9731-8d5e1c28a1b9.png)

**Comment by theaspiringitpro** [comment]
Here's a link to the profiles. [https://drive.google.com/file/d/1xkUdpbB7oYBxH8vgL7FpK2_PeD4s_p8m/view?usp=sharing](url)

**Comment by ledoge** [comment]
Hmmm... I can reproduce the red tint, but I'm not sure if this is a bug with novideo_srgb or a problem with the profile, as the gamut view looks kinda strange: 
![image](https://user-images.githubusercontent.com/8040316/150609955-622a973d-019f-4297-8490-66f8f9d8093e.png)

Can you try making a new profile?

**Comment by theaspiringitpro** [comment]
TBH, I'm fairly happy with my current profile and would rather not have to chase 6500k again. Took me several attempts to find the right OSD settings and they seem to drift or change drastically each time I calibrate.

Here's a link to the measurement report I performed after calibration if that helps.

[https://drive.google.com/file/d/1IHLUWSvxC9rXtyDrXmD9aE07Lh4DpdZt/view?usp=sharing](url)

**Comment by theaspiringitpro** [comment]
Decided to scrap the old profile. The new calibration turned out near perfect. Hit 6501k, along with a slightly higher contrast ratio.

Here's a link to the new profile and measurement report. [https://drive.google.com/file/d/1j8oLGXiUQSaH9wN-JJtfq8x9MqQSsiRC/view?usp=sharing](url)

The profile in gamut view now looks normal and I'm no longer seeing a red hue when using your tool. 

![screenshot](https://user-images.githubusercontent.com/16659780/150619279-a493665e-4d59-4846-8809-5c89f819d7ee.png)

Seems like XYZ + matrix profiles are now on the table bois! Really appreciate the troubleshooting help. 

Now if I could only figure out what caused that red hue anomaly... lol

**Comment by ledoge** [comment]
I realized that my implementation was flawed – it worked fine only with some profiles, specifically ones that had (near-) linear input shaper curves for the XYZ LUT. Wrote some more code that should hopefully do the lookups properly now. Please test your old "weird" profile again with the new v2.1 release and see if it results in accurate colors now. From my testing, at least the red tint is gone.

**Comment by aufkrawall** [comment]
Seems to be looking fine with the profile in XYZ format for my previous monitor, black output offset seems to behave as expected and I also don't notice any tint. Are there visual differences to be expected vs. curve format, e.g. because tone response curve starts at (0, 0) instead of actual black point?

**Comment by DeBloba** [comment]
Hey, I don't know if it is the profile (found it online, as I have no idea how any of this works) or the software, but with v2.1 I suddenly get a very blue-greenish tint on everything.

My profile: 
https://1drv.ms/u/s!Au8o_YnLfndsjxpEVsM8RIbBYzki?e=t7C3dE

**Comment by theaspiringitpro** [comment]
> I realized that my implementation was flawed – it worked fine only with some profiles, specifically ones that had (near-) linear input shaper curves for the XYZ LUT. Wrote some more code that should hopefully do the lookups properly now. Please test your old "weird" profile again with the new v2.1 release and see if it results in accurate colors now. From my testing, at least the red tint is gone.

The changes you made seem to have worked with the old profile. Not seeing a redish tint anymore.

**Comment by ledoge** [comment]
>Are there visual differences to be expected vs. curve format, e.g. because tone response curve starts at (0, 0) instead of actual black point?

Not entirely sure how the black point compensation in profiles created with DisplayCAL works, but if I had to guess, I would expect the resulting grayscale response to be similar to using 100% output offset, with some very minor differences in color reproduction.

Also, I realized that my CLUT implementation was *still* wrong (at least compared to ArgyllCMS, but the ICC standard doesn't specify how the values should actually be encoded...). With v2.2, the lookups now produce 100% the same results as with ArgyllCMS, so everything *should* work fine now.
